core/actions/types.ts

```diff
@@ -58,7 +65,7 @@ export type Action<
 	P extends ActionPubType = ActionPubType,
 	C extends ZodObjectOrWrapped = ZodObjectOrWrapped,
 	A extends ZodObjectOrWrappedOrOptional = ZodObjectOrWrappedOrOptional,
-	N extends string = string,
+	N extends ActionName = ActionName,
```

Comment:
Because of `strict: true` enabling [`strictFunctionTypes`](https://www.typescriptlang.org/tsconfig/#strictFunctionTypes), we often ran into the issue that we were doing something like this:

```ts
import type { Action as ActionName } from 'db/public'

// newly defined function which uses the Enum to narrow action names
const addActionFn = (action: ActionName) => { /* ... */ }

// needs to be string
const onAdd = (action: string) => { /*...*/ }

export function AddButton = ({action, onAdd}: {
	action: Action // the whole thing, not just the name
	onAdd: (actionName: string) => void // needs to take actionname: string bc Action['name'] was `string`
}) => {
		return <Button onClick={()=>onAdd(action.name)} > Add {action.name} </Button>
}
```

Then, when you wanted to use the more correct `actionFn` which uses the `ActionName` enum as arg, like so

```tsx
<AddButton onAdd={addActionFn} action={action} />
```
You  would get an error with `strict: true`, bc with `strictFunctionTypes` the function arguments need to match more closely.

I think this restriction is in general good, see the linked Typescript page for examples of errors.

To solve this, I thought it was nicest to just require actions to use the `Action` enum generated from the `db`, as those are the only allowed options anyway.